### ISSUE TITLE
Can't add email address to a project via bulk upload

### ISSUE THREAD

**Issue by seledwards** [GENI-NSF/geni-portal#713]
Niky tried to add (via bulk upload) an email of the form:
  first.last@EXAMPLE.EDU

It didn't work.  Why?

 _Imported from trac ticket [#713](http://trac.gpolab.bbn.com/proto-ch/ticket/713),  created by sedwards on 07-15-2013 at 20:07, last modified: 01-08-2014 at 13:20_
   CCing: nriga@...

**Comment by ahelsing** [comment]
I think I had the same problem in the scaling test. My guess is a problem in the comparing of addresses case insensitively - the address is uppercase in the database, so if we don't lowercase both the address in the database and the address uploaded, it will appear the user has no account.

_Trac comment by ahelsing on 07-15-2013 at 20:13_

**Comment by seledwards** [comment]
I think I tried both lowercase and uppercase versions of her email when uploading and neither worked (but I'd have to double check).  

_Trac comment by sedwards (github user: seledwards) on 07-16-2013 at 09:49_

**Comment by ahelsing** [comment]
upload-project-members lowercases the email address

Then the list of email addresses is passed to lookup_members_by_email

That method in the ma_controller does a strict == search for existing email address.

So no matter what case you use in the file, if the email address is not all lowercase in our database, it will not be found.

To fix this, lookup_members_by_email should lowercase the email_address in the DB that it is comparing, as well as lowercasing the email addressed passed in in the call (to make sure we don't care if all callers lowercased the arg).

IE ma_controller line 962
```
  foreach($email_args as $email) {
    if ($emails != "") $emails = $emails . ", ";
    $emails = $emails . "'" . strtolower($email) . "'";
  }
  $sql = "select " . MA_ATTRIBUTE::MEMBER_ID . ", " . MA_ATTRIBUTE::VALUE . 
    " from " . $MA_MEMBER_ATTRIBUTE_TABLENAME . 
    " where " . MA_ATTRIBUTE::NAME . " = '" . MA_ATTRIBUTE_NAME::EMAIL_ADDRESS . "'" . 
    " and lower(" . MA_ATTRIBUTE::VALUE . ") in (" . $emails . ")";
```




_Trac comment by ahelsing on 07-16-2013 at 10:06_

**Comment by seledwards** [comment]
I found several more instances of this in the list of GEC17 tutorial attendees.


_Trac comment by sedwards (github user: seledwards) on 07-16-2013 at 18:41_

**Comment by seledwards** [comment]
We just experienced the exact same issue as described in this ticket in preparing for Winter Camp 2014 except the emails had capital letters in the username portions of their email like this: FirstLast@example.edu

We saw this issue with two email addresses.  Reopening.

_Trac comment by sedwards (github user: seledwards) on 01-06-2014 at 15:30_

**Comment by ahelsing** [comment]
This may require a chapi fix. EG MAv1Implementation around line 220 may need to notice that the attribute is email and add logic to treat the value as case insensitive.

_Trac comment by ahelsing on 01-07-2014 at 09:00_

**Comment by ahelsing** [comment]
That may not be the right fix, since technically (RFC 2821) the name part of the email address is supposed to be case sensitive. Probably instead the client should be comparing things case sensitive for the name part and insensitive for the domain part. Put another way, the error you saw may be the right behavior.

_Trac comment by ahelsing on 01-07-2014 at 09:33_

**Comment by seledwards** [comment]
Our CSV file has a case sensitive email (`FirstLast@example.edu`) but the comparison fails and portal page displays a lowercase version of the email (`firstlast@example.edu`).

This is definitely not right.

A case sensitive comparison might be ok, but then the portal needs to not lowercase it first.

_Trac comment by sedwards (github user: seledwards) on 01-07-2014 at 10:28_

**Comment by ahelsing** [comment]
This will be resolved by chapi ticket #203

_Trac comment by ahelsing on 01-08-2014 at 13:20_
